### PR TITLE
Make static build

### PR DESCRIPTION
**Pull request by brybacki**
Signed-off-by: Bartosz Rybacki <brybacki@redhat.com>

**What this PR does / why we need it**:

Make sure binary is statically linked. Simplify the build and make it similar to other plugins (like for example velero-plugin-for-csi).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Binary is statically linked
```

### CODE REVIEW

File: hack/build/build-functest.sh [hunk 1]
```diff
@@ -14,13 +14,29 @@
...
 
-set -e
+set -euo pipefail
 
-script_dir="$(cd "$(dirname "$0")" && pwd -P)"
-source "${script_dir}"/../config.sh
+export PATH=$PATH:$HOME/gopath/bin
+JOB_TYPE="${JOB_TYPE:-}"
+
+if [ "${JOB_TYPE}" == "travis" ]; then
```
Comment (awels) [inline hack/build/build-functest.sh:22]: travis was disabled completely for the kubevirt organization, so you can probably simplify this if else

**Review by awels — COMMENTED** [review]
looks good, one minor comment

File: hack/build/build-functest.sh (hunk 1 above)
Comment (brybacki) [reply hack/build/build-functest.sh]: I was looking at HCO script while doing this. 

That is great. I've removed that part

**Pull request merged**